src/execution_plan/ops/op_distinct.c

```diff
@@ -0,0 +1,49 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "op_distinct.h"
+#include "xxhash/xxhash.h"
+#include "util/arr.h" 
+#include "util/triemap/triemap.h" 
+
+OpBase* NewDistinctOp(ResultSet *resultset) {
+    OpDistinct *self = malloc(sizeof(OpDistinct));
+    self->trie = NewTrieMap();
+
+    OpBase_Init(&self->op);
+    self->op.name = "Distinct";
+    self->op.type = OPType_DISTINCT;
+    self->op.consume = DistinctConsume;
+    self->op.reset = DistinctReset;
+    self->op.free = DistinctFree;
+
+    return (OpBase*)self;
+}
+
+Record DistinctConsume(OpBase *opBase) {
+    OpDistinct *self = (OpDistinct*)opBase;
+    OpBase *child = self->op.children[0];
+
+    while(true) {
```

Comment:
Wouldn't you prefer:
while((r = child->consume(child))) {
    unsigned long long const hash = Record_Hash64(r); ....
}

Reply:
Well, I'm not sure whether the following variant is actually clearer than the original one.

Record r;
while((r = child->consume(child))) {
    unsigned long long const hash = Record_Hash64(r);
    int is_new = TrieMap_Add(self->trie, (char *) &hash, sizeof(hash), NULL, TrieMap_DONT_CARE_REPLACE);
    if(is_new) return r;
    Record_Free(r);
}
return NULL;